### PR TITLE
storage/gc_worker: Add more metrics

### PR DESCRIPTION
**Pull request by MyonKeminta**
## What have you changed? (mandatory)

This PR adds two metrics to gc_worker.
* `AUTO_GC_PROCESSED_REGIONS_GAUGE_VEC`: counts number of scanned/gc-ed regions for each round of GC. Resets to zero when the round of GC finishes or rewinding occurs.
* `AUTO_GC_SAFE_POINT_GAUGE`: The current safe point of the TiKV instance.

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Please describe the tests that you ran to verify your changes. Have you finished unit tests, integration tests, or manual tests? What additional tests would give you greater confidence in this change?

## Does this PR affect documentation (docs) or release note? (mandatory)

- No

## Does this PR affect tidb-ansible update? (mandatory)

- Yes

### CODE REVIEW

**Comment by MyonKeminta** [comment]
@zhangjinpeng1987 PTAL. The example of these metrics are here: https://github.com/pingcap/tidb-ansible/pull/761

File: src/storage/gc_worker.rs [hunk 1]
```diff
@@ -779,6 +789,8 @@ impl<S: GCSafePointProvider, R: RegionInfoProvider> GCManager<S, R> {
             }
         };
 
+        AUTO_GC_SAFE_POINT_GAUGE.set(safe_point as i64);
```
Comment (AndreMouche) [inline src/storage/gc_worker.rs]: I think set it after we update the `self.safe_point` will be better?

**Comment by MyonKeminta** [comment]
@AndreMouche PTAL

**Comment by youjiali1995** [comment]
```
error: unneeded return statement

   --> src/storage/gc_worker.rs:810:9

    |

810 |         return updated;

    |         ^^^^^^^^^^^^^^^ help: remove `return` as shown: `updated`

    |

```

**Comment by MyonKeminta** [comment]
@AndreMouche PTAL again thanks!

File: src/storage/gc_worker.rs [hunk 2]
```diff
@@ -793,7 +803,9 @@ impl<S: GCSafePointProvider, R: RegionInfoProvider> GCManager<S, R> {
                 self.safe_point = safe_point;
```
Comment (AndreMouche) [inline src/storage/gc_worker.rs]: I mean set the metrics after L793

**Pull request merged**